tests/unit/deferred.py

```diff
@@ -0,0 +1,201 @@
+# Copyright 2016 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+'''Functional test for deferred.py'''
+
+import pytest
+import os
+import tempfile
+
+from hil import config, deferred, model
+from hil.model import db, Switch
+from hil.test_common import config_testsuite, config_merge, \
+                             fresh_database, fail_on_log_warnings
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+fail_on_log_warnings = pytest.fixture(autouse=True)(fail_on_log_warnings)
+fresh_database = pytest.fixture(fresh_database)
+
+INTERFACE1 = '104/0/10'
+INTERFACE2 = '104/0/18'
+
+DeferredTestSwitch = None
+temp_db = tempfile.NamedTemporaryFile()
+
+
+@pytest.fixture
+def configure():
+    config_testsuite()
+
+    additional_config = {
+        'extensions': {
+            'hil.ext.obm.mock': ''
+            }
+        }
+
+    # if we are using sqlite's in memory db, then change uri to a db on disk
+    uri = config.cfg.get('database', 'uri')
+    if uri == 'sqlite:///:memory:':
+        uri = 'sqlite:///' + temp_db.name
```

Comment:
In that case I had the yield inside the if statement, and if that branch is not taken (i.e. running tests with postgres), the tests error out with the following error:
`.venv/lib/python2.7/site-packages/_pytest/fixtures.py:699: StopIteration`

```python
    if uri == 'sqlite:///:memory:':
        with tempfile.NamedTemporaryFile() as temp_db:
            uri = 'sqlite:///' + temp_db.name
            additional_config['database'] = {'uri': uri}
            config_merge(additional_config)
            config.load_extensions()
            yield
    else:
        config_merge(additional_config)
        config.load_extensions()
```

If i get rid of yield, the db is still available to the rest of the code and is cleaned up on exit.